OutOfSchool/OutOfSchool.DataAccess/Repository/PictureStorageMongo.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using MongoDB.Bson;
+using MongoDB.Driver;
+using MongoDB.Driver.GridFS;
+using OutOfSchool.Services.Configuration;
+
+namespace OutOfSchool.Services.Repository
+{
+    public class PictureStorageMongo : IPictureStorage
+    {
+        private readonly MongoRepositoryConfiguration storageConfiguration;
+
+        public PictureStorageMongo(IOptions<MongoRepositoryConfiguration> storageConfiguration)
+        {
+            this.storageConfiguration = storageConfiguration?.Value
+                ?? throw new ArgumentNullException(nameof(storageConfiguration));
+        }
+
+        /// <summary>
+        /// Returns picture by picture Id.
+        /// </summary>
+        /// <param name="pictureId">Picture id.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public async Task<Stream> GetPictureByIdAsync(string pictureId)
+            // Note: await should be here bacause of type conversion: Task<GridFSDownloadStream> cannot be automaticaly cast to the Task<Stream> ...
+            => await GetGreedFSBucket().OpenDownloadStreamAsync(new ObjectId(pictureId));
+
+        private IGridFSBucket GetGreedFSBucket()
+        {
+            // TODO: check performance
+            var client = new MongoClient(storageConfiguration.ConnectionString);
```

Comment:
not sure if it's ok to create a new client for every request as it will probably create a new connection to Mongo (just a question, as i don't know .Net Mongo client implementation, maybe it's fine).